docs/Developer Documentation/Logger/index.md

```diff
@@ -0,0 +1,159 @@
+# SDL Logger
+
+Open source version of SDL is using log4cxx for logging.
+Because of lack of log4cxx stability OEM have an ability to replace log4cxx logger with any other logging boost or syslog.
+
+
+SDL implements Logger abstration for easily replace library for logging.
+
+|||
+High Level Design
+![Logger](./assets/high_level_design.png)
+|||
+
+
+All components have aceess to `Logger` interface and use it for logging.
+
+## SDLLogger
+
+`Logger` interface is implemented by `SDLLoggerImpl`. 
+
+SDLLoggerImpl use message loop thread to proxy loggin message to thard party (external) logger.
+SDLLoggerImpl owns `ExternalLogger` and controlls it's life cycle. 
+SDLLoggerImpl provide implementation of the singleton pattern.
+
+
+### Message loop thread in SDLLogger
+
+Message loop thread is needed to avoid significant performance degradation in run time is logging calls are blocking calls and takes to much time. 
+`SDLLoggerImpl::PushLog` is non blocking call. It will put log message in to the queue and exit immediately 
+
+
+If `ExternalLogger` supports non blocking threaded logging, minor changes in SDLLogger are required : `SDLLoggerImpl::PushLog` should be reimplemented to 
+use `ExternalLogger::ForceLog()` directly. 
+
+## Logger singleton 
+
+Logger is the only one singleton class in SDL.
+Singleton pattern required to have an access to logger from any component. 
+Logger singleton provides singleton by `Logger` interface. 
+So sdl components do not have information about neither logger implementation nor specific external logger. 
+
+## Logger singleton with plugins 
+
+SDL plugins are shared librariess, so LoggerSingleton could not be implemented with Mayers singleton. 
+Mayers singleton would create own SDL logger instance.
+
+The idea is to pass singleton pointer to Plugin during creation, so that plugin could initialize Logger::instance pointer with received from sdl_core. 
```

Comment:
```suggestion
The idea is to pass singleton pointer to Plugin during creation, so that plugin could initialize `Logger::instance` pointer with one received from SDL core. 
```